---
title: "Explorative Data Analysis of DCs infected with Yersinia scRNA-Seq data"
author: "Health Data Unit"
date: "`r format(Sys.time(), '%d %B, %Y')`"
output: 
  html_document:
    code_folding: hide
---


This report provides documentation for the analysis of scRNA-Seq data of Dendritic Cells 
progenitors infected with *Yersinia enterocolitica* provided by the Department of 
Internal Medicine II, University of TÃ¼bingen, Germany.


## Data Analysis

A repository storing custom scripts has been created and is available in github 
[here](https://github.com/caramirezal/interaction_DC-pathogens). This repository
along with the data are stored locally in the BioQuant curry cluster of the 
University of Heidelberg.

Data were preprocessed as shown in the `reports/scPipe_QC_workflow.Rmd` file.

```{r setup, include=FALSE}
## Dependencies
library(dplyr)
library(scPipe)
library(ggplot2)
library(rmarkdown)
library(Seurat)
library(ggrepel)

knitr::opts_chunk$set(
        warning = FALSE, 
        message = FALSE,
        cache = FALSE,
        fig.align='center'
)

## Setting paths
path2project <- '/media/ag-cherrmann/cramirez/interaction_DC-pathogens/'
knitr::opts_knit$set(
        root.dir = path2project
)
```

# Demultiplexed counts

The folowing table shows annotations and quality metrics for all the wells in the 
dataset. 

```{r, settings}
## Loading data
seurat <- readRDS('data/yersinia_seurat.rds')

## Reading processed annotations
anns <- readxl::read_xlsx('data/20201214 C025_sample_sheet_SEA.xlsx') %>%
            as.data.frame()
rownames(anns) <- with(anns, paste0(plate_number, '_', well_position))
## Adding annotations
anns <- anns[rownames(seurat@meta.data), ]
anns <- anns[, c("progenitor_type: 'CDP' or 'MDP'",
                 "bacteria: 'HK Ye' or 'Ye' or 'no Ye'",
                 "treatment: A-untreated; B-MOI 1; C-MOI 0.1; D-MOI 0.01",
                 "bacteria_attached: '+' or '-'",
                 'infection_time')]
colnames(anns)[1:4] <- c('progenitor', 'bacteria', 'treatment', 'bacteria_attached')
## row order check
#any(rownames(anns) != rownames(seurat@meta.data))
seurat@meta.data <- cbind(seurat@meta.data, anns)

## Selected category
category <- 'plate_number'
```

```{r, results='asis'}
seurat@meta.data %>%
        paged_table()
```

# Single cell analysis

First, we focused in the analysis of single cells. Then, we only selected wells containing
single cells which correspond to plate 193, 194 and 196 as can be seen in the next table. 


```{r, single_cells_table, results='asis'}
table(seurat$sample_name, seurat$plate_number) %>%
        as.data.frame.matrix() %>%
        paged_table()
```

## Filtering cells


We decided to filter out single cells with less than 50 number of RNA counts as 
visualized in the next scatter plot of number of RNA counts *vs* the percentage of
mitochondrial genes. The vertical dashed red line shows the threshold to filter
out cells. Cells below the threshold were filtered out.


```{r, filtering}
sc.seurat <- subset(seurat, sample_name=='single cell')
counts_th <- 50

## Number of RNA counts vs % mito
sc.seurat@meta.data %>% 
  ggplot(aes(x=log(nCount_RNA), 
             colour=get(category),
             y=percent.mt)) +
        geom_point(alpha=0.5) +
        labs(x='Log(Counts of RNA)', y='Percentage mitochondrial',
             colour='') +
        geom_vline(xintercept = log(counts_th), 
                   linetype = 'dashed',
                   colour='red') +
        theme_classic()

sc.seurat <- subset(sc.seurat, nCount_RNA > counts_th)
```

After filtering `r ncol(sc.seurat)`/`r ncol(subset(seurat, sample_name=='single cell'))`
single cells passed the quality control. 


```{r, single_cells_umap, fig.align='center'}
sc.seurat <- NormalizeData(sc.seurat) %>% ScaleData()
sc.seurat <- FindVariableFeatures(sc.seurat)
sc.seurat <- RunPCA(sc.seurat, 
                    npcs = 50, 
                    features = VariableFeatures(sc.seurat))
sc.seurat <- RunUMAP(sc.seurat, 
                     reduction = 'pca', 
                     dims = 1:20)
```

The next UMAP projections show single cells according to the different categories 
in clusters. Four clusters can be observed which coincides with cells in plates. 
Some cells of plate 196 (blue) fall more close to that of plates 193 (salmon) and 194 (green). This suggest the existence of a batch effect ought to differences in 
sequencing of different plates. 


```{r categories_in_single_cells, fig.align='center', fig.width=10, fig.height=11}
## Annotating with yersinia infection and cell type
sc.seurat$treatment[is.na(sc.seurat$treatment)] <- 'NA'
plate <- sc.seurat %>%
        DimPlot(label = TRUE, 
                label.box = TRUE, 
                repel = TRUE) + NoLegend() +
        ggtitle('Plate')

disease.state <- sc.seurat %>%
        DimPlot(group.by = c('disease_state'),
                label = TRUE,
                label.box = TRUE) + 
        NoLegend() +
        ggtitle('Disease State')

bacteria <- sc.seurat %>%
        DimPlot(group.by = c('bacteria_attached'),
                label = TRUE,
                label.box = TRUE, 
                repel = TRUE) + 
        NoLegend() +
        ggtitle('Bacteria')

treatment <- sc.seurat %>%
        DimPlot(group.by = c('treatment'),
                label = TRUE,
                label.box = TRUE, 
                repel = TRUE) + 
        NoLegend() +
        ggtitle('Treatment')

progenitor <- sc.seurat %>%
        DimPlot(group.by = c('progenitor'),
                label = TRUE,
                label.box = TRUE, 
                repel = TRUE) + 
        NoLegend() +
        ggtitle('Progenitor')

infection_time <- sc.seurat %>%
        DimPlot(group.by = c('infection_time'),
                label = TRUE,
                label.box = TRUE, 
                repel = TRUE) + 
        NoLegend() +
        ggtitle('Infection time')

gridExtra::grid.arrange(
  grobs = list(plate, disease.state, 
               bacteria, treatment, 
               progenitor, infection_time)
)
```

# Differential expression analysis


### Definition of Bystander Cells

The following table shows the definition of bystander cells according to treatment and
whether the cell has Yersinia attached.


```{r}
sc.seurat$'treated' <- ifelse(sc.seurat$treatment == 'MOI 0', 'Mock-Infected', 'Infected')
sc.seurat$'Bystander' <- paste0(sc.seurat$treated, '_', sc.seurat$bacteria_attached) 
sc.seurat$Bystander <- gsub(' ', '_', sc.seurat$Bystander)
sc.seurat$Bystander <- plyr::mapvalues(
  sc.seurat$Bystander,
  from = c('Infected_-', 'Infected_+', 'Mock-Infected_-'),
  to = c('Bystander', 'Infected', 'Mock-Infected')
)
sc.seurat@meta.data %>%
  select(treatment, bacteria_attached, Bystander) %>%
  table() %>%
  as.data.frame() %>%
  filter(Freq > 0) %>%
  arrange(treatment) %>%
  paged_table()
```

Using these definitions we performed a differential expression analysis taking cells 
in plates separately.

The next table show the frequency of single cells according to their
infection status and the type of progenitor.

```{r}
sc.seurat@meta.data %>%
  filter(plate_number=='P193') %>%
  select(Bystander, progenitor) %>%
  table() %>%
  as.data.frame() %>%
  filter(Freq>0) %>%
  paged_table()
```

## 1 HPI (Plate 193)


## Vulcano plots

DEGs are shown in the following vulcano plots for CDP and MDP at time 1 hpi.
Some top genes based on fold change expression with p-val < 0.05 are shown
in black and labeled in each comparison. Full tables with all genes 
with significant change are provided at the end of this report.


```{r cdp_1hr_vulcano_plots}
Idents(sc.seurat) <- as.character(sc.seurat$Bystander)
infvsBys.cdp.deg <- sc.seurat %>%
  subset(progenitor=='CDP' & plate_number=='P193') %>%
  subset(Bystander != 'Mock-Infected') %>%
  FindMarkers(ident.1='Infected') %>%
  arrange(desc(abs(avg_log2FC)))
infvsBys.cdp.plot <- infvsBys.cdp.deg %>%
  add_rownames('gene') %>%
  filter(!grepl('^Gm', gene)) %>%
  mutate(highlight=ifelse( ( abs(avg_log2FC) > 0.75  & p_val < 0.05),
                           TRUE, FALSE)) %>%
  mutate(label=ifelse(highlight==TRUE, gene, '')) %>%
  ggplot(aes(x=avg_log2FC, y=-log10(p_val),
             colour=highlight,
             label=label)) +
        geom_point(show.legend = FALSE) +
        geom_text_repel(show.legend=FALSE) +
        theme(legend.position = 'none') +
        scale_color_manual(values = c('salmon', 'black')) +
        theme_classic() +
        xlim(-2, 2) +
        ggtitle('CDP - 1HPI - Infected vs Bystander')


infvsMock.cdp.deg <- sc.seurat %>%
  subset(progenitor=='CDP' & plate_number=='P193') %>%
  subset(Bystander != 'Bystander') %>%
  FindMarkers(ident.1='Infected') %>%
  arrange(desc(abs(avg_log2FC)))
infvsMock.cdp.plot <- infvsMock.cdp.deg %>%  
  add_rownames('gene') %>%
  filter(!grepl('^Gm', gene)) %>%
  mutate(highlight=ifelse( ( abs(avg_log2FC) > 0.8  & p_val < 0.05),
                           TRUE, FALSE)) %>%
  mutate(label=ifelse(highlight==TRUE, gene, '')) %>%
  ggplot(aes(x=avg_log2FC, y=-log10(p_val),
             colour=highlight,
             label=label)) +
        geom_point(show.legend = FALSE) +
        geom_text_repel(show.legend=FALSE) +
        theme(legend.position = 'none') +
        scale_color_manual(values = c('salmon', 'black')) +
        theme_classic() +
        xlim(-2, 2) +
        ggtitle('CDP - 1HPI - Infected vs Mock-Infected')


bysvsMock.cdp.deg <- sc.seurat %>%
  subset(progenitor=='CDP' & plate_number=='P193') %>%
  subset(Bystander != 'Infected') %>%
  FindMarkers(ident.1='Bystander') %>%
  arrange(desc(abs(avg_log2FC)))
bysvsMock.cdp.plot <- bysvsMock.cdp.deg %>%  
  add_rownames('gene') %>%
  filter(!grepl('^Gm', gene)) %>%
  mutate(highlight=ifelse( ( abs(avg_log2FC) > 0.5  & p_val < 0.05),
                           TRUE, FALSE)) %>%
  mutate(label=ifelse(highlight==TRUE, gene, '')) %>%
  ggplot(aes(x=avg_log2FC, y=-log10(p_val),
             colour=highlight,
             label=label)) +
        geom_point(show.legend = FALSE) +
        geom_text_repel(show.legend=FALSE) +
        theme(legend.position = 'none') +
        scale_color_manual(values = c('salmon', 'black')) +
        theme_classic() +
        xlim(-2, 2) +
        ggtitle('CDP - 1HPI - Bystander vs Mock-Infected')
```

```{r mdp_1hr_vulcano_plots}
Idents(sc.seurat) <- as.character(sc.seurat$Bystander)
infvsBys.mdp.deg <- sc.seurat %>%
  subset(progenitor=='MDP' & plate_number=='P193') %>%
  subset(Bystander != 'Mock-Infected') %>%
  FindMarkers(ident.1='Infected') %>%
  arrange(desc(abs(avg_log2FC)))
infvsBys.mdp.plot <- infvsBys.mdp.deg %>%
  add_rownames('gene') %>%
  filter(!grepl('^Gm', gene)) %>%
  mutate(highlight=ifelse( ( abs(avg_log2FC) > 0.9  & p_val < 0.05),
                           TRUE, FALSE)) %>%
  mutate(label=ifelse(highlight==TRUE, gene, '')) %>%
  ggplot(aes(x=avg_log2FC, y=-log10(p_val),
             colour=highlight,
             label=label)) +
        geom_point(show.legend = FALSE) +
        geom_text_repel(show.legend=FALSE) +
        theme(legend.position = 'none') +
        scale_color_manual(values = c('salmon', 'black')) +
        theme_classic() +
        xlim(-2, 2) +
        ggtitle('MDP - 1HPI - Infected vs Bystander')


infvsMock.mdp.deg <- sc.seurat %>%
  subset(progenitor=='MDP' & plate_number=='P193') %>%
  subset(Bystander != 'Bystander') %>%
  FindMarkers(ident.1='Infected') %>%
  arrange(desc(abs(avg_log2FC)))
infvsMock.mdp.plot <- infvsMock.mdp.deg %>%  
  add_rownames('gene') %>%
  filter(!grepl('^Gm', gene)) %>%
  mutate(highlight=ifelse( ( abs(avg_log2FC) > 0.85  & p_val < 0.05),
                           TRUE, FALSE)) %>%
  mutate(label=ifelse(highlight==TRUE, gene, '')) %>%
  ggplot(aes(x=avg_log2FC, y=-log10(p_val),
             colour=highlight,
             label=label)) +
        geom_point(show.legend = FALSE) +
        geom_text_repel(show.legend=FALSE) +
        theme(legend.position = 'none') +
        scale_color_manual(values = c('salmon', 'black')) +
        theme_classic() +
        xlim(-2, 2) +
        ggtitle('MDP - 1HPI - Infected vs Mock-Infected')


bysvsMock.mdp.deg <- sc.seurat %>%
  subset(progenitor=='MDP' & plate_number=='P193') %>%
  subset(Bystander != 'Infected') %>%
  FindMarkers(ident.1='Bystander') %>%
  arrange(desc(abs(avg_log2FC)))
bysvsMock.mdp.plot <- bysvsMock.mdp.deg %>%  
  add_rownames('gene') %>%
  filter(!grepl('^Gm', gene)) %>%
  mutate(highlight=ifelse( ( abs(avg_log2FC) > 0.9  & p_val < 0.05),
                           TRUE, FALSE)) %>%
  mutate(label=ifelse(highlight==TRUE, gene, '')) %>%
  ggplot(aes(x=avg_log2FC, y=-log10(p_val),
             colour=highlight,
             label=label)) +
        geom_point(show.legend = FALSE) +
        geom_text_repel(show.legend=FALSE) +
        theme(legend.position = 'none') +
        scale_color_manual(values = c('salmon', 'black')) +
        theme_classic() +
        xlim(-2, 2) +
        ggtitle('MDP - 1HPI - Bystander vs Mock-Infected')
```


```{r, fig.width=15, fig.height=10}
gridExtra::grid.arrange(
  grobs=list(infvsBys.cdp.plot,
             infvsMock.cdp.plot, 
             bysvsMock.cdp.plot,
             infvsBys.mdp.plot,
             infvsMock.mdp.plot,
             bysvsMock.mdp.plot),
  ncol=3
)
```

## 24 HPI (Plate 196)


DEGs are shown in the following vulcano plots for CDP and MDP at time 24 hpi.
Some top genes based on fold change expression with p-val < 0.05 are shown
in black and labeled in each comparison. Full tables with all genes 
with significant change are provided at the end of this report.


```{r cdp_24hr_vulcano_plots}
Idents(sc.seurat) <- as.character(sc.seurat$Bystander)
infvsBys.cdp.deg.24 <- sc.seurat %>%
  subset(progenitor=='CDP' & plate_number=='P196') %>%
  subset(Bystander != 'Mock-Infected') %>%
  FindMarkers(ident.1='Infected') %>%
  arrange(desc(abs(avg_log2FC)))
infvsBys.cdp.plot.24 <- infvsBys.cdp.deg.24 %>%
  add_rownames('gene') %>%
  filter(!grepl('^Gm', gene)) %>%
  mutate(highlight=ifelse( ( avg_log2FC < - 0.85  & p_val < 0.05 ) |
                          ( avg_log2FC > 2  & p_val < 0.05 ), TRUE, FALSE)) %>%
  mutate(label=ifelse(highlight==TRUE, gene, '')) %>%
  ggplot(aes(x=avg_log2FC, y=-log10(p_val),
             colour=highlight,
             label=label)) +
        geom_point(show.legend = FALSE) +
        geom_text_repel(show.legend=FALSE) +
        theme(legend.position = 'none') +
        scale_color_manual(values = c('salmon', 'black')) +
        theme_classic() +
        xlim(-2, 3) +
        ggtitle('CDP - 24HPI - Infected vs Bystander')


infvsMock.cdp.deg.24 <- sc.seurat %>%
  subset(progenitor=='CDP' & plate_number=='P196') %>%
  subset(Bystander != 'Bystander') %>%
  FindMarkers(ident.1='Infected') %>%
  arrange(desc(abs(avg_log2FC)))
infvsMock.cdp.plot.24 <- infvsMock.cdp.deg.24 %>%  
  add_rownames('gene') %>%
  filter(!grepl('^Gm', gene)) %>%
  mutate(highlight=ifelse( ( avg_log2FC < - 0.5  & p_val < 0.05) |
                             ( avg_log2FC > 2  & p_val < 0.05 ),
                           TRUE, FALSE)) %>%
  mutate(label=ifelse(highlight==TRUE, gene, '')) %>%
  ggplot(aes(x=avg_log2FC, y=-log10(p_val),
             colour=highlight,
             label=label)) +
        geom_point(show.legend = FALSE) +
        geom_text_repel(show.legend=FALSE) +
        theme(legend.position = 'none') +
        scale_color_manual(values = c('salmon', 'black')) +
        theme_classic() +
        xlim(-4, 4) +
        ggtitle('CDP - 24HPI - Infected vs Mock-Infected')

bysvsMock.cdp.deg.24 <- sc.seurat %>%
  subset(progenitor=='CDP' & plate_number=='P196') %>%
  subset(Bystander != 'Infected') %>%
  FindMarkers(ident.1='Bystander') %>%
  arrange(desc(abs(avg_log2FC)))
bysvsMock.cdp.plot.24 <- bysvsMock.cdp.deg.24 %>%  
  add_rownames('gene') %>%
  filter(!grepl('^Gm', gene)) %>%
  mutate(highlight=ifelse( ( abs(avg_log2FC) > 1  & p_val < 0.05),
                           TRUE, FALSE)) %>%
  mutate(label=ifelse(highlight==TRUE, gene, '')) %>%
  ggplot(aes(x=avg_log2FC, y=-log10(p_val),
             colour=highlight,
             label=label)) +
        geom_point(show.legend = FALSE) +
        geom_text_repel(show.legend=FALSE, force = 3) +
        theme(legend.position = 'none') +
        scale_color_manual(values = c('salmon', 'black')) +
        theme_classic() +
        xlim(-2, 3) +
        ggtitle('CDP - 24HPI - Bystander vs Mock-Infected')
```


```{r mdp_24hr_vulcano_plots}
Idents(sc.seurat) <- as.character(sc.seurat$Bystander)
infvsBys.mdp.deg.24 <- sc.seurat %>%
  subset(progenitor=='MDP' & plate_number=='P196') %>%
  subset(Bystander != 'Mock-Infected') %>%
  FindMarkers(ident.1='Infected') %>%
  arrange(desc(abs(avg_log2FC)))
infvsBys.mdp.plot.24 <- infvsBys.mdp.deg.24 %>%
  add_rownames('gene') %>%
  filter(!grepl('^Gm', gene)) %>%
  mutate(highlight=ifelse( ( abs(avg_log2FC) > 0.9  & p_val < 0.05),
                           TRUE, FALSE)) %>%
  mutate(label=ifelse(highlight==TRUE, gene, '')) %>%
  ggplot(aes(x=avg_log2FC, y=-log10(p_val),
             colour=highlight,
             label=label)) +
        geom_point(show.legend = FALSE) +
        geom_text_repel(show.legend=FALSE) +
        theme(legend.position = 'none') +
        scale_color_manual(values = c('salmon', 'black')) +
        theme_classic() +
        xlim(-2, 2) +
        ggtitle('MDP - 24HPI - Infected vs Bystander')


infvsMock.mdp.deg.24 <- sc.seurat %>%
  subset(progenitor=='MDP' & plate_number=='P196') %>%
  subset(Bystander != 'Bystander') %>%
  FindMarkers(ident.1='Infected') %>%
  arrange(desc(abs(avg_log2FC)))
infvsMock.mdp.plot.24 <- infvsMock.mdp.deg.24 %>%  
  add_rownames('gene') %>%
  filter(!grepl('^Gm', gene)) %>%
  mutate(highlight=ifelse( ( avg_log2FC < -1  & p_val < 0.05 ) |
                             ( avg_log2FC > 1.75  & p_val < 0.05 ),
                           TRUE, FALSE)) %>%
  mutate(label=ifelse(highlight==TRUE, gene, '')) %>%
  ggplot(aes(x=avg_log2FC, y=-log10(p_val),
             colour=highlight,
             label=label)) +
        geom_point(show.legend = FALSE) +
        geom_text_repel(show.legend=FALSE) +
        theme(legend.position = 'none') +
        scale_color_manual(values = c('salmon', 'black')) +
        theme_classic() +
        xlim(-3, 3) +
        ggtitle('MDP - 24HPI - Infected vs Mock-Infected')


bysvsMock.mdp.deg.24 <- sc.seurat %>%
  subset(progenitor=='MDP' & plate_number=='P196') %>%
  subset(Bystander != 'Infected') %>%
  FindMarkers(ident.1='Bystander') %>%
  arrange(desc(abs(avg_log2FC)))
bysvsMock.mdp.plot.24 <- bysvsMock.mdp.deg.24 %>%  
  add_rownames('gene') %>%
  filter(!grepl('^Gm', gene)) %>%
  mutate(highlight=ifelse( ( abs(avg_log2FC) > 1.1  & p_val < 0.05),
                           TRUE, FALSE)) %>%
  mutate(label=ifelse(highlight==TRUE, gene, '')) %>%
  ggplot(aes(x=avg_log2FC, y=-log10(p_val),
             colour=highlight,
             label=label)) +
        geom_point(show.legend = FALSE) +
        geom_text_repel(show.legend=FALSE) +
        theme(legend.position = 'none') +
        scale_color_manual(values = c('salmon', 'black')) +
        theme_classic() +
        xlim(-4.5, 4.5) +
        ggtitle('MDP - 24HPI - Bystander vs Mock-Infected')
```


```{r, fig.width=15, fig.height=10}
gridExtra::grid.arrange(
  grobs=list(infvsBys.cdp.plot.24,
             infvsMock.cdp.plot.24, 
             bysvsMock.cdp.plot.24,
             infvsBys.mdp.plot.24,
             infvsMock.mdp.plot.24,
             bysvsMock.mdp.plot.24),
  ncol=3
)
```


# Additional tables

The next tables show all the genes in the DEAs with p-val < 0.05.

## 1 HPI

### CDPs. 

* Table comparing Infected *vs* Bystander

```{r table_cdp_inf_vs_bys_1hpi}
infvsBys.cdp.deg %>%
  filter(p_val < 0.05) %>%
  paged_table()
```

* Table comparing Bystander *vs* Mock-Infected

```{r}
bysvsMock.cdp.deg %>%
  filter(p_val < 0.05) %>%
  paged_table()
```


* Table comparing Infected *vs* Mock-Infected


```{r}
infvsMock.cdp.deg %>%
  filter(p_val < 0.05) %>%
  paged_table()
```


### MDPs. 

* Table comparing Infected *vs* Bystander

```{r table_mdp_inf_vs_bys_1hpi}
infvsBys.mdp.deg %>%
  filter(p_val < 0.05) %>%
  paged_table()
```

* Table comparing Bystander *vs* Mock-Infected

```{r}
bysvsMock.mdp.deg %>%
  filter(p_val < 0.05) %>%
  paged_table()
```


* Table comparing Infected *vs* Mock-Infected

```{r}
infvsMock.mdp.deg %>%
  filter(p_val < 0.05) %>%
  paged_table()
```


## 24 HPI


### CDPs. 

* Table comparing Infected *vs* Bystander

```{r table_cdp_inf_vs_bys_24hpi}
infvsBys.cdp.deg.24 %>%
  filter(p_val < 0.05) %>%
  paged_table()
```

* Table comparing Bystander *vs* Mock-Infected

```{r}
bysvsMock.cdp.deg.24 %>%
  filter(p_val < 0.05) %>%
  paged_table()
```


* Table comparing Infected *vs* Mock-Infected


```{r}
infvsMock.cdp.deg.24 %>%
  filter(p_val < 0.05) %>%
  paged_table()
```


### MDPs. 

* Table comparing Infected *vs* Bystander

```{r table_mdp_inf_vs_bys_24hpi}
infvsBys.mdp.deg.24 %>%
  filter(p_val < 0.05) %>%
  paged_table()
```

* Table comparing Bystander *vs* Mock-Infected

```{r}
bysvsMock.mdp.deg.24 %>%
  filter(p_val < 0.05) %>%
  paged_table()
```


* Table comparing Infected *vs* Mock-Infected

```{r}
infvsMock.mdp.deg.24 %>%
  filter(p_val < 0.05) %>%
  paged_table()
```
