---
title: "Explorative Data Analysis of DCs infected with Yersinia scRNA-Seq data"
author: "Health Data Unit"
date: "`r format(Sys.time(), '%d %B, %Y')`"
output:
  html_document:
    code_folding: hide
  pdf_document: default
---


This report provides documentation for the analysis of scRNA-Seq data of Dendritic Cells 
progenitors infected with *Yersinia enterocolitica* provided by the Department of 
Internal Medicine II, University of TÃ¼bingen, Germany.


## Data Analysis

A repository storing custom scripts has been created and is available in github 
[here](https://github.com/caramirezal/interaction_DC-pathogens). This repository
along with the data are stored locally in the BioQuant curry cluster of the 
University of Heidelberg.

Data were preprocessed as shown in the `reports/scPipe_QC_workflow.Rmd` file.

```{r setup, include=FALSE}
## Dependencies
library(scPipe)
library(ggplot2)
library(rmarkdown)
library(Seurat)
library(ggrepel)
library(viridis)
library(destiny)
library(enrichR)
library(dplyr)
library(reshape2)

set.seed(333)

knitr::opts_chunk$set(
        warning = FALSE, 
        message = FALSE,
        cache = FALSE,
        fig.align='center'
)

## Setting paths
path2project <- '/media/ag-cherrmann/cramirez/interaction_DC-pathogens/'
knitr::opts_knit$set(
        root.dir = path2project
)
```

# Demultiplexed counts

The following table shows annotations and quality metrics for all the wells in the
dataset. 

```{r, settings}
## Loading data
seurat <- readRDS('data/yersinia_seurat.rds')

## Reading processed annotations
anns <- readxl::read_xlsx('data/20201214 C025_sample_sheet_SEA.xlsx') %>%
            as.data.frame()
rownames(anns) <- with(anns, paste0(plate_number, '_', well_position))
## Adding annotations
anns <- anns[rownames(seurat@meta.data), ]
anns <- anns[, c("progenitor_type: 'CDP' or 'MDP'",
                 "bacteria: 'HK Ye' or 'Ye' or 'no Ye'",
                 "treatment: A-untreated; B-MOI 1; C-MOI 0.1; D-MOI 0.01",
                 "bacteria_attached: '+' or '-'",
                 'infection_time')]
colnames(anns)[1:4] <- c('progenitor', 'bacteria', 'treatment', 'bacteria_attached')
## row order check
#any(rownames(anns) != rownames(seurat@meta.data))
seurat@meta.data <- cbind(seurat@meta.data, anns)

## Selected category
category <- 'plate_number'
```

```{r, results='asis'}
seurat@meta.data %>%
        paged_table()
```

# Single cell analysis

First, we focused in the analysis of single cells. Then, we only selected wells containing
single cells which correspond to plate 193, 194 and 196 as can be seen in the next table. 


```{r, single_cells_table, results='asis'}
table(seurat$sample_name, seurat$plate_number) %>%
        as.data.frame.matrix() %>%
        paged_table()
```

## QC metrics



```{r, comparisson_old_new_metrics, fig.width=14}
new.qc <- seurat@meta.data %>%
  filter(sample_name == 'single cell') %>%
  mutate(plate=gsub('_.*', '', well)) %>%
    dplyr::select(plate, nCount_RNA, 
                  nFeature_RNA, 
                  unaligned:mapped_to_MT) %>%
    melt() %>%
    filter(variable!='mapped_to_ERCC') %>%
    ggplot(aes(x=plate, y=log10(value), fill=plate)) +
          geom_violin() +
          facet_wrap(~variable) +
          theme_bw() +
          theme(legend.position = 'none') +
          labs(x='', y='Log10(Counts)') +
          ggtitle('New demultiplexing')

old.demultiplexing <- readRDS('data/DCs-pathogen_interactions_seu.rds')
old.qc <- old.demultiplexing@meta.data %>%
  add_rownames(var = 'well') %>%
  filter(sample_name == 'single cell') %>%
    dplyr::select(plate_number, nCount_RNA, 
                  nFeature_RNA, 
                  unaligned:mapped_to_MT) %>%
    melt() %>%
    filter(variable!='mapped_to_ERCC') %>%
    ggplot(aes(x=plate_number, y=log10(value), fill=plate_number)) +
          geom_violin() +
          facet_wrap(~variable) +
          theme_bw() +
          theme(legend.position = 'none') +
          labs(x='', y='Log10(Counts)') +
          ggtitle('Old demultiplexing')

gridExtra::grid.arrange(
  new.qc,
  old.qc,
  ncol=2
)
```

## Filtering cells


We decided to filter out single cells with less than 50 number of RNA counts as 
visualized in the next scatter plot of number of RNA counts *vs* the percentage of
mitochondrial genes. The vertical dashed red line shows the threshold to filter
out cells. Cells below the threshold were filtered out.


```{r, filtering}
sc.seurat <- subset(seurat, sample_name=='single cell')
counts_th <- 50

## Number of RNA counts vs % mito
sc.seurat@meta.data %>% 
  ggplot(aes(x=log(nCount_RNA), 
             colour=get(category),
             y=percent.mt)) +
        geom_point(alpha=0.5) +
        labs(x='Log(Counts of RNA)', y='Percentage mitochondrial',
             colour='') +
        geom_vline(xintercept = log(counts_th), 
                   linetype = 'dashed',
                   colour='red') +
        theme_classic()

sc.seurat <- subset(sc.seurat, nCount_RNA > counts_th)
```

After filtering `r ncol(sc.seurat)`/`r ncol(subset(seurat, sample_name=='single cell'))`
single cells passed the quality control. 


```{r, single_cells_umap, fig.align='center'}
sc.seurat <- NormalizeData(sc.seurat) %>% ScaleData()
sc.seurat <- FindVariableFeatures(sc.seurat)
sc.seurat <- RunPCA(sc.seurat, 
                    npcs = 50, 
                    features = VariableFeatures(sc.seurat))
sc.seurat <- RunUMAP(sc.seurat, 
                     reduction = 'pca', 
                     dims = 1:20)
```

The next UMAP projections show single cells according to the different categories 
in clusters. Four clusters can be observed which coincides with cells in plates. 
Some cells of plate 196 (blue) fall more close to that of plates 193 (salmon) and 194 (green). This suggest the existence of a batch effect ought to differences in 
sequencing of different plates. 


```{r categories_in_single_cells, fig.align='center', fig.width=10, fig.height=11}
## Annotating with yersinia infection and cell type
sc.seurat$treatment[is.na(sc.seurat$treatment)] <- 'NA'
plate <- sc.seurat %>%
        DimPlot(label = TRUE, 
                label.box = TRUE, 
                repel = TRUE) + NoLegend() +
        ggtitle('Plate')

disease.state <- sc.seurat %>%
        DimPlot(group.by = c('disease_state'),
                label = TRUE,
                label.box = TRUE) + 
        NoLegend() +
        ggtitle('Disease State')

bacteria <- sc.seurat %>%
        DimPlot(group.by = c('bacteria_attached'),
                label = TRUE,
                label.box = TRUE, 
                repel = TRUE) + 
        NoLegend() +
        ggtitle('Bacteria')

treatment <- sc.seurat %>%
        DimPlot(group.by = c('treatment'),
                label = TRUE,
                label.box = TRUE, 
                repel = TRUE) + 
        NoLegend() +
        ggtitle('Treatment')

progenitor <- sc.seurat %>%
        DimPlot(group.by = c('progenitor'),
                label = TRUE,
                label.box = TRUE, 
                repel = TRUE) + 
        NoLegend() +
        ggtitle('Progenitor')

infection_time <- sc.seurat %>%
        DimPlot(group.by = c('infection_time'),
                label = TRUE,
                label.box = TRUE, 
                repel = TRUE) + 
        NoLegend() +
        ggtitle('Infection time')

gridExtra::grid.arrange(
  grobs = list(plate, disease.state, 
               bacteria, treatment, 
               progenitor, infection_time)
)
```

# Differential expression analysis


### Definition of Bystander Cells

The following table shows the definition of bystander cells according to treatment and
whether the cell has Yersinia attached.


```{r}
sc.seurat$'treated' <- ifelse(sc.seurat$treatment == 'MOI 0', 
                              'Mock-Infected', 'Infected')
sc.seurat$'Bystander' <- paste0(sc.seurat$treated, '_', sc.seurat$bacteria_attached) 
sc.seurat$Bystander <- gsub(' ', '_', sc.seurat$Bystander)
sc.seurat$Bystander <- plyr::mapvalues(
  sc.seurat$Bystander,
  from = c('Infected_-', 'Infected_+', 'Mock-Infected_-'),
  to = c('Bystander', 'Infected', 'Mock-Infected')
)
sc.seurat@meta.data %>%
  dplyr::select(treatment, bacteria_attached, Bystander) %>%
  table() %>%
  as.data.frame() %>%
  filter(Freq > 0) %>%
  arrange(treatment) %>%
  paged_table()
```

Using these definitions we performed a differential expression analysis taking cells 
in plates separately.

The next table show the frequency of single cells according to their
infection status and the type of progenitor in plate 193.

### Plate 193

```{r, plate_193_composition}
sc.seurat@meta.data %>%
  filter(plate_number=='P193') %>%
  dplyr::select(Bystander, progenitor) %>%
  table() %>%
  as.data.frame() %>%
  filter(Freq>0) %>%
  paged_table()
```


### Plate 196


```{r, plate_196_composition}
sc.seurat@meta.data %>%
  filter(plate_number=='P196') %>%
  dplyr::select(Bystander, progenitor) %>%
  table() %>%
  as.data.frame() %>%
  filter(Freq>0) %>%
  paged_table()
```


## Vulcano plots

## 1 HPI (Plate 193)


DEGs are shown in the following vulcano plots for CDP and MDP at time 1 hpi.
Some top genes based on fold change expression with p-val < 0.05 are shown
in black and labeled in each comparison. Full tables with all genes 
with significant change are provided at the end of this report.


```{r cdp_1hr_vulcano_plots}
## Loading selected genes
selected_genes <- readLines('data/selected_genes.txt')
#selected_genes

Idents(sc.seurat) <- as.character(sc.seurat$Bystander)
infvsBys.cdp.deg <- sc.seurat %>%
  subset(progenitor=='CDP' & plate_number=='P193') %>%
  subset(Bystander != 'Mock-Infected') %>%
  FindMarkers(ident.1='Infected') %>%
  arrange(desc(abs(avg_log2FC))) %>%
  add_rownames('gene') %>%
  mutate(selected_gene=(toupper(gene) %in% toupper(selected_genes))) %>%
  filter(!grepl('^Gm', gene)) %>%
  mutate(highlight=ifelse( ( abs(avg_log2FC) > 1  & p_val < 0.05 ),
                           TRUE, FALSE) ) %>%
  mutate(label=ifelse(highlight==TRUE | selected_gene ==TRUE, gene, ''))
infvsBys.cdp.plot <- infvsBys.cdp.deg %>%
  ggplot(aes(x=avg_log2FC, y=-log10(p_val),
             colour=highlight,
             label=label)) +
        geom_point(show.legend = FALSE) +
        geom_point(data=subset(infvsBys.cdp.deg,
                               selected_gene==TRUE), 
                   aes(colour='green'),
                   show.legend=FALSE) +
        geom_text_repel(show.legend=FALSE, 
                        colour='black') +
        theme(legend.position = 'none') +
        scale_color_manual(values = c('salmon', 'green', 
                                      'black', 'red')) +
        theme_classic() +
        xlim(-2, 2) +
        ggtitle('CDP - 1HPI - Infected vs Bystander')
#infvsBys.cdp.plot

infvsMock.cdp.deg <- sc.seurat %>%
  subset(progenitor=='CDP' & plate_number=='P193') %>%
  subset(Bystander != 'Bystander') %>%
  FindMarkers(ident.1='Infected') %>%
  arrange(desc(abs(avg_log2FC))) %>%
  add_rownames('gene') %>%
  mutate(selected_gene=(toupper(gene) %in% toupper(selected_genes))) %>%
  filter(!grepl('^Gm', gene)) %>%
  mutate(highlight=ifelse( ( abs(avg_log2FC) > 1  & p_val < 0.05 ),
                           TRUE, FALSE) ) %>%
  mutate(label=ifelse(highlight==TRUE | selected_gene ==TRUE, gene, ''))
infvsMock.cdp.plot <- infvsMock.cdp.deg %>%  
  ggplot(aes(x=avg_log2FC, y=-log10(p_val),
             colour=highlight,
             label=label)) +
        geom_point(show.legend = FALSE) +
        geom_point(data=subset(infvsMock.cdp.deg,
                               selected_gene==TRUE), 
                   aes(colour='green'),
                   show.legend=FALSE) +
        geom_text_repel(show.legend=FALSE, 
                        colour='black') +
        theme(legend.position = 'none') +
        scale_color_manual(values = c('salmon', 'green', 
                                      'black', 'red')) +
        theme_classic() +
        xlim(-2, 2) +
        ggtitle('CDP - 1HPI - Infected vs Mock-Infected')
#infvsMock.cdp.plot


bysvsMock.cdp.deg <- sc.seurat %>%
  subset(progenitor=='CDP' & plate_number=='P193') %>%
  subset(Bystander != 'Infected') %>%
  FindMarkers(ident.1='Bystander') %>%
  arrange(desc(abs(avg_log2FC))) %>%
  add_rownames('gene') %>%
  mutate(selected_gene=(toupper(gene) %in% toupper(selected_genes))) %>%
  filter(!grepl('^Gm', gene)) %>%
  mutate(highlight=ifelse( ( abs(avg_log2FC) > 0.5  & p_val < 0.05 ),
                           TRUE, FALSE) ) %>%
  mutate(label=ifelse(highlight==TRUE | selected_gene ==TRUE, gene, ''))
bysvsMock.cdp.plot <- bysvsMock.cdp.deg %>%  
  ggplot(aes(x=avg_log2FC, y=-log10(p_val),
             colour=highlight,
             label=label)) +
        geom_point(show.legend = FALSE) +
        geom_point(data=subset(bysvsMock.cdp.deg,
                               selected_gene==TRUE), 
                   aes(colour='green'),
                   show.legend=FALSE) +
        geom_text_repel(show.legend=FALSE, 
                        colour='black') +
        theme(legend.position = 'none') +
        scale_color_manual(values = c('salmon', 'green', 
                                      'black', 'red')) +
        theme_classic() +
        xlim(-2, 2) +
        ggtitle('CDP - 1HPI - Bystander vs Mock-Infected')
```


```{r mdp_1hr_vulcano_plots}
#Idents(sc.seurat) <- as.character(sc.seurat$Bystander)
infvsBys.mdp.deg <- sc.seurat %>%
  subset(progenitor=='MDP' & plate_number=='P193') %>%
  subset(Bystander != 'Mock-Infected') %>%
  FindMarkers(ident.1='Infected') %>%
  arrange(desc(abs(avg_log2FC))) %>%
  add_rownames('gene') %>%
  mutate(selected_gene=(toupper(gene) %in% toupper(selected_genes))) %>%
  filter(!grepl('^Gm', gene)) %>%
  mutate(highlight=ifelse( ( abs(avg_log2FC) > 0.9  & p_val < 0.05),
                           TRUE, FALSE) ) %>%
  mutate(label=ifelse(highlight==TRUE | selected_gene ==TRUE, gene, ''))
infvsBys.mdp.plot <- infvsBys.mdp.deg %>%
  ggplot(aes(x=avg_log2FC, y=-log10(p_val),
             colour=highlight,
             label=label)) +
        geom_point(show.legend = FALSE) +
        geom_point(data=subset(infvsBys.mdp.deg,
                               selected_gene==TRUE), 
                   aes(colour='green'),
                   show.legend=FALSE) +
        geom_text_repel(show.legend=FALSE, 
                        colour='black') +
        theme(legend.position = 'none') +
        scale_color_manual(values = c('salmon', 'green', 
                                      'black', 'red')) +
        theme_classic() +
        xlim(-2, 2) +
        ggtitle('MDP - 1HPI - Infected vs Bystander')

infvsMock.mdp.deg <- sc.seurat %>%
  subset(progenitor=='MDP' & plate_number=='P193') %>%
  subset(Bystander != 'Bystander') %>%
  FindMarkers(ident.1='Infected') %>%
  arrange(desc(abs(avg_log2FC))) %>%
  add_rownames('gene') %>%
  mutate(selected_gene=(toupper(gene) %in% toupper(selected_genes))) %>%
  filter(!grepl('^Gm', gene)) %>%
  mutate(highlight=ifelse( ( abs(avg_log2FC) > 0.85  & p_val < 0.05),
                           TRUE, FALSE) ) %>%
  mutate(label=ifelse(highlight==TRUE | selected_gene ==TRUE, gene, ''))
infvsMock.mdp.plot <- infvsMock.mdp.deg %>%  
  ggplot(aes(x=avg_log2FC, y=-log10(p_val),
             colour=highlight,
             label=label)) +
        geom_point(show.legend = FALSE) +
        geom_point(data=subset(infvsMock.mdp.deg,
                               selected_gene==TRUE), 
                   aes(colour='green'),
                   show.legend=FALSE) +
        geom_text_repel(show.legend=FALSE, 
                        colour='black') +
        theme(legend.position = 'none') +
        scale_color_manual(values = c('salmon', 'green', 
                                      'black', 'red')) +
        theme_classic() +
        xlim(-2, 2) +
        ggtitle('MDP - 1HPI - Infected vs Mock-Infected')


bysvsMock.mdp.deg <- sc.seurat %>%
  subset(progenitor=='MDP' & plate_number=='P193') %>%
  subset(Bystander != 'Infected') %>%
  FindMarkers(ident.1='Bystander') %>%
  arrange(desc(abs(avg_log2FC))) %>%
  add_rownames('gene') %>%
  mutate(selected_gene=(toupper(gene) %in% toupper(selected_genes))) %>%
  filter(!grepl('^Gm', gene)) %>%
  mutate(highlight=ifelse( ( abs(avg_log2FC) > 0.9  & p_val < 0.05 ),
                           TRUE, FALSE) ) %>%
  mutate(label=ifelse(highlight==TRUE | selected_gene ==TRUE, gene, ''))
bysvsMock.mdp.plot <- bysvsMock.mdp.deg %>%  
  ggplot(aes(x=avg_log2FC, y=-log10(p_val),
             colour=highlight,
             label=label)) +
        geom_point(show.legend = FALSE) +
        geom_point(data=subset(bysvsMock.mdp.deg,
                               selected_gene==TRUE), 
                   aes(colour='green'),
                   show.legend=FALSE) +
        geom_text_repel(show.legend=FALSE, 
                        colour='black') +
        theme(legend.position = 'none') +
        scale_color_manual(values = c('salmon', 'green', 
                                      'black', 'red')) +
        theme_classic() +
        xlim(-2, 2) +
        ggtitle('MDP - 1HPI - Bystander vs Mock-Infected')

```


```{r, fig.width=15, fig.height=10}
gridExtra::grid.arrange(
  grobs=list(infvsBys.cdp.plot,
             infvsMock.cdp.plot, 
             bysvsMock.cdp.plot,
             infvsBys.mdp.plot,
             infvsMock.mdp.plot,
             bysvsMock.mdp.plot),
  ncol=3
)
```

## 24 HPI (Plate 196)


DEGs are shown in the following vulcano plots for CDP and MDP at time 24 hpi.
Some top genes based on fold change expression with p-val < 0.05 are shown
in black and labeled in each comparison. Full tables with all genes 
with significant change are provided at the end of this report.


```{r cdp_24hr_vulcano_plots}
Idents(sc.seurat) <- as.character(sc.seurat$Bystander)
infvsBys.cdp.deg.24 <- sc.seurat %>%
  subset(progenitor=='CDP' & plate_number=='P196') %>%
  subset(Bystander != 'Mock-Infected') %>%
  FindMarkers(ident.1='Infected') %>%
  arrange(desc(abs(avg_log2FC))) %>%
  add_rownames('gene') %>%
  mutate(selected_gene=(toupper(gene) %in% toupper(selected_genes))) %>%
  filter(!grepl('^Gm', gene)) %>%
  mutate(highlight=ifelse( ( avg_log2FC < - 0.85  & p_val < 0.05 ) |
                          ( avg_log2FC > 2  & p_val < 0.05 ),
                           TRUE, FALSE) ) %>%
  mutate(label=ifelse(highlight==TRUE | selected_gene ==TRUE, gene, ''))
infvsBys.cdp.plot.24 <- infvsBys.cdp.deg.24 %>%
  ggplot(aes(x=avg_log2FC, y=-log10(p_val),
             colour=highlight,
             label=label)) +
        geom_point(show.legend = FALSE) +
        geom_point(data=subset(infvsBys.cdp.deg.24,
                               selected_gene==TRUE), 
                   aes(colour='green'),
                   show.legend=FALSE) +
        geom_text_repel(show.legend=FALSE, 
                        colour='black') +
        theme(legend.position = 'none') +
        scale_color_manual(values = c('salmon', 'green', 
                                      'black', 'red')) +
        theme_classic() +
        xlim(-2, 3) +
        ggtitle('CDP - 24HPI - Infected vs Bystander')


infvsMock.cdp.deg.24 <- sc.seurat %>%
  subset(progenitor=='CDP' & plate_number=='P196') %>%
  subset(Bystander != 'Bystander') %>%
  FindMarkers(ident.1='Infected') %>%
  arrange(desc(abs(avg_log2FC))) %>%
  add_rownames('gene') %>%
  mutate(selected_gene=(toupper(gene) %in% toupper(selected_genes))) %>%
  filter(!grepl('^Gm', gene)) %>%
  mutate(highlight=ifelse( ( avg_log2FC < - 0.5  & p_val < 0.05 ) |
                             ( avg_log2FC > 2  & p_val < 0.05 ),
                           TRUE, FALSE) ) %>%
  mutate(label=ifelse(highlight==TRUE | selected_gene ==TRUE, gene, ''))
infvsMock.cdp.plot.24 <- infvsMock.cdp.deg.24 %>%  
  ggplot(aes(x=avg_log2FC, y=-log10(p_val),
             colour=highlight,
             label=label)) +
        geom_point(show.legend = FALSE) +
        geom_point(data=subset(infvsMock.cdp.deg.24,
                               selected_gene==TRUE), 
                   aes(colour='green'),
                   show.legend=FALSE) +
        geom_text_repel(show.legend=FALSE, 
                        colour='black') +
        theme(legend.position = 'none') +
        scale_color_manual(values = c('salmon', 'green', 
                                      'black', 'red')) +
        theme_classic() +
        xlim(-4, 4) +
        ggtitle('CDP - 24HPI - Infected vs Mock-Infected')

bysvsMock.cdp.deg.24 <- sc.seurat %>%
  subset(progenitor=='CDP' & plate_number=='P196') %>%
  subset(Bystander != 'Infected') %>%
  FindMarkers(ident.1='Bystander') %>%
  arrange(desc(abs(avg_log2FC))) %>%
  add_rownames('gene') %>%
  mutate(selected_gene=(toupper(gene) %in% toupper(selected_genes))) %>%
  filter(!grepl('^Gm', gene)) %>%
  mutate(highlight=ifelse( ( abs(avg_log2FC) > 1  & p_val < 0.05 ),
                           TRUE, FALSE) ) %>%
  mutate(label=ifelse(highlight==TRUE | selected_gene ==TRUE, gene, ''))
bysvsMock.cdp.plot.24 <- bysvsMock.cdp.deg.24 %>%  
  ggplot(aes(x=avg_log2FC, y=-log10(p_val),
             colour=highlight,
             label=label)) +
        geom_point(show.legend = FALSE) +
        geom_point(data=subset(bysvsMock.cdp.deg.24,
                               selected_gene==TRUE), 
                   aes(colour='green'),
                   show.legend=FALSE) +
        geom_text_repel(show.legend=FALSE, 
                        colour='black', 
                        force = 3) +
        theme(legend.position = 'none') +
        scale_color_manual(values = c('salmon', 'green', 
                                      'black', 'red')) +
        theme_classic() +
        xlim(-2, 3) +
        ggtitle('CDP - 24HPI - Bystander vs Mock-Infected')
```


```{r mdp_24hr_vulcano_plots}
#Idents(sc.seurat) <- as.character(sc.seurat$Bystander)
infvsBys.mdp.deg.24 <- sc.seurat %>%
  subset(progenitor=='MDP' & plate_number=='P196') %>%
  subset(Bystander != 'Mock-Infected') %>%
  FindMarkers(ident.1='Infected') %>%
  arrange(desc(abs(avg_log2FC))) %>%
  add_rownames('gene') %>%
  mutate(selected_gene=(toupper(gene) %in% toupper(selected_genes))) %>%
  filter(!grepl('^Gm', gene)) %>%
  mutate(highlight=ifelse( ( abs(avg_log2FC) > 0.9  & p_val < 0.05),
                           TRUE, FALSE) ) %>%
  mutate(label=ifelse(highlight==TRUE | selected_gene ==TRUE, gene, ''))
infvsBys.mdp.plot.24 <- infvsBys.mdp.deg.24 %>%
  ggplot(aes(x=avg_log2FC, y=-log10(p_val),
             colour=highlight,
             label=label)) +
        geom_point(show.legend = FALSE) +
        geom_point(data=subset(infvsBys.mdp.deg.24,
                               selected_gene==TRUE), 
                   aes(colour='green'),
                   show.legend=FALSE) +
        geom_text_repel(show.legend=FALSE, 
                        colour='black', 
                        force = 3.5) +
        theme(legend.position = 'none') +
        scale_color_manual(values = c('salmon', 'green', 
                                      'black', 'red')) +
        theme_classic() +
        xlim(-2, 2) +
        ggtitle('MDP - 24HPI - Infected vs Bystander')


infvsMock.mdp.deg.24 <- sc.seurat %>%
  subset(progenitor=='MDP' & plate_number=='P196') %>%
  subset(Bystander != 'Bystander') %>%
  FindMarkers(ident.1='Infected') %>%
  arrange(desc(abs(avg_log2FC))) %>%
  add_rownames('gene') %>%
  mutate(selected_gene=(toupper(gene) %in% toupper(selected_genes))) %>%
  filter(!grepl('^Gm', gene)) %>%
  mutate(highlight=ifelse( ( avg_log2FC < -1  & p_val < 0.05 ) |
                             ( avg_log2FC > 1.75  & p_val < 0.05 ),
                           TRUE, FALSE) ) %>%
  mutate(label=ifelse(highlight==TRUE | selected_gene ==TRUE, gene, ''))
infvsMock.mdp.plot.24 <- infvsMock.mdp.deg.24 %>%  
  ggplot(aes(x=avg_log2FC, y=-log10(p_val),
             colour=highlight,
             label=label)) +
        geom_point(show.legend = FALSE) +
        geom_point(data=subset(infvsMock.mdp.deg.24,
                               selected_gene==TRUE), 
                   aes(colour='green'),
                   show.legend=FALSE) +
        geom_text_repel(show.legend=FALSE, 
                        colour='black', 
                        force = 3.5) +
        theme(legend.position = 'none') +
        scale_color_manual(values = c('salmon', 'green', 
                                      'black', 'red')) +
        theme_classic() +
        xlim(-2, 2) +
        ggtitle('MDP - 24HPI - Infected vs Mock-Infected')


bysvsMock.mdp.deg.24 <- sc.seurat %>%
  subset(progenitor=='MDP' & plate_number=='P196') %>%
  subset(Bystander != 'Infected') %>%
  FindMarkers(ident.1='Bystander') %>%
  arrange(desc(abs(avg_log2FC))) %>%
  add_rownames('gene') %>%
  mutate(selected_gene=(toupper(gene) %in% toupper(selected_genes))) %>%
  filter(!grepl('^Gm', gene)) %>%
  mutate(highlight=ifelse( ( abs(avg_log2FC) > 1.1  & p_val < 0.05 ),
                           TRUE, FALSE) ) %>%
  mutate(label=ifelse(highlight==TRUE | selected_gene ==TRUE, gene, ''))
bysvsMock.mdp.plot.24 <- bysvsMock.mdp.deg.24 %>%  
  ggplot(aes(x=avg_log2FC, y=-log10(p_val),
             colour=highlight,
             label=label)) +
        geom_point(show.legend = FALSE) +
        geom_point(data=subset(bysvsMock.mdp.deg.24,
                               selected_gene==TRUE), 
                   aes(colour='green'),
                   show.legend=FALSE) +
        geom_text_repel(show.legend=FALSE, 
                        colour='black', 
                        force = 3.5) +
        theme(legend.position = 'none') +
        scale_color_manual(values = c('salmon', 'green', 
                                      'black', 'red')) +
        theme_classic() +
        xlim(-4.5, 4.5) +
        ggtitle('MDP - 24HPI - Bystander vs Mock-Infected')
```


```{r, fig.width=15, fig.height=10}
gridExtra::grid.arrange(
  grobs=list(infvsBys.cdp.plot.24,
             infvsMock.cdp.plot.24, 
             bysvsMock.cdp.plot.24,
             infvsBys.mdp.plot.24,
             infvsMock.mdp.plot.24,
             bysvsMock.mdp.plot.24),
  ncol=3
)
```

## Markers expression

The next dotplots show gene expression of the top 15 with highest positive FC and the same
for the negative (summing up 30 genes) and for which p-vals are less than 0.05. 

### 1 hpi (Plate 193)

```{r, fig.height=16, fig.width=16}
degs <- list(infvsBys.cdp.deg=infvsBys.cdp.deg, 
             infvsMock.cdp.deg=infvsMock.cdp.deg, 
             bysvsMock.cdp.deg=bysvsMock.cdp.deg,
             infvsBys.mdp.deg=infvsBys.mdp.deg, 
             infvsMock.mdp.deg=infvsMock.mdp.deg, 
             bysvsMock.mdp.deg=bysvsMock.mdp.deg,
             infvsBys.cdp.deg.24=infvsBys.cdp.deg.24, 
             infvsMock.cdp.deg.24=infvsMock.cdp.deg.24, 
             bysvsMock.cdp.deg.24=bysvsMock.cdp.deg.24,
             infvsBys.mdp.deg.24=infvsBys.mdp.deg.24, 
             infvsMock.mdp.deg.24=infvsMock.mdp.deg.24, 
             bysvsMock.mdp.deg.24=bysvsMock.mdp.deg.24)
degs.filtered <- lapply(degs, function(x) filter(x, p_val < 0.05) %>%
                          arrange(desc(avg_log2FC)))

n_markers <- 15

# CDP - 1HPI - Infected vs Bystander
infvsBys.cdp.1hpi.hmap <- DoHeatmap(
  object = subset(sc.seurat, 
                  progenitor=='CDP' & 
                    plate_number=='P193' & 
                    Bystander != 'Mock-Infected'), 
  features = c(head(degs.filtered$infvsBys.cdp.deg$gene, n_markers),
               tail(degs.filtered$infvsBys.cdp.deg$gene, n_markers)), 
  draw.lines = FALSE, 
  size = 3, 
  angle = 0, combine = TRUE
) + scale_fill_viridis() +
  theme(legend.position = 'none') +
  ggtitle('CDP - 1HPI - Infected vs Bystander')


# CDP - 1HPI - Infected vs Mock
infvsMock.cdp.1hpi.hmap <- DoHeatmap(
  object = subset(sc.seurat, 
                  progenitor=='CDP' & 
                    plate_number=='P193' & 
                    Bystander != 'Bystander'), 
  features = c(head(degs.filtered$infvsMock.cdp.deg$gene, n_markers),
               tail(degs.filtered$infvsMock.cdp.deg$gene, n_markers)), 
  draw.lines = FALSE, 
  size = 3, 
  angle = 0, combine = TRUE
) + scale_fill_viridis() +
  theme(legend.position = 'none') +
  ggtitle('CDP - 1HPI - Infected vs Mock')

##############################################################################################

## CDP - 1HPI - Infected vs Bystander
infvsBys.cdp.1hpi.dotplot <- DotPlot(
  object = subset(sc.seurat, 
                  progenitor=='CDP' & 
                    plate_number=='P193' & 
                    Bystander != 'Mock-Infected'), 
  features = c(head(degs.filtered$infvsBys.cdp.deg$gene, n_markers),
               tail(degs.filtered$infvsBys.cdp.deg$gene, n_markers))
) + coord_flip() +
  scale_color_viridis() +
  labs(x='', y='') +
  theme(axis.text.x = element_text(angle = 45, hjust = 1)) +
  ggtitle('CDP - 1HPI - Infected vs Bystander')


## CDP - 1HPI - Infected vs Mock-Infected
infvsMock.cdp.1hpi.dotplot <- DotPlot(
  object = subset(sc.seurat, 
                  progenitor=='CDP' & 
                    plate_number=='P193' & 
                    Bystander != 'Bystander'), 
  features = c(head(degs.filtered$infvsMock.cdp.deg$gene, n_markers),
               tail(degs.filtered$infvsMock.cdp.deg$gene, n_markers))
) + coord_flip() +
  scale_color_viridis() +
  labs(x='', y='') +
  theme(axis.text.x = element_text(angle = 45, hjust = 1)) +
  ggtitle('CDP - 1HPI - Infected vs Mock')


## CDP - 1HPI - Bystander vs Mock-Infected
BysvsMock.cdp.1hpi.dotplot <- DotPlot(
  object = subset(sc.seurat, 
                  progenitor=='CDP' & 
                    plate_number=='P193' & 
                    Bystander != 'Infected'), 
  features = c(head(degs.filtered$bysvsMock.cdp.deg$gene, n_markers),
               tail(degs.filtered$bysvsMock.cdp.deg$gene, n_markers))
) + coord_flip() +
  scale_color_viridis() +
  labs(x='', y='') +
  theme(axis.text.x = element_text(angle = 45, hjust = 1)) +
  ggtitle('CDP - 1HPI - Infected vs Mock')

###################################################################################

## CDP - 1HPI - Infected vs Bystander
infvsBys.mdp.1hpi.dotplot <- DotPlot(
  object = subset(sc.seurat, 
                  progenitor=='MDP' & 
                    plate_number=='P193' & 
                    Bystander != 'Mock-Infected'), 
  features = c(head(degs.filtered$infvsBys.mdp.deg$gene, n_markers),
               tail(degs.filtered$infvsBys.mdp.deg$gene, n_markers))
) + coord_flip() +
  scale_color_viridis() +
  labs(x='', y='') +
  theme(axis.text.x = element_text(angle = 45, hjust = 1)) +
  ggtitle('MDP - 1HPI - Infected vs Bystander')


## CDP - 1HPI - Infected vs Mock-Infected
infvsMock.mdp.1hpi.dotplot <- DotPlot(
  object = subset(sc.seurat, 
                  progenitor=='MDP' & 
                    plate_number=='P193' & 
                    Bystander != 'Bystander'), 
  features = c(head(degs.filtered$infvsMock.mdp.deg$gene, n_markers),
               tail(degs.filtered$infvsMock.mdp.deg$gene, n_markers))
) + coord_flip() +
  scale_color_viridis() +
  labs(x='', y='') +
  theme(axis.text.x = element_text(angle = 45, hjust = 1)) +
  ggtitle('MDP - 1HPI - Infected vs Mock')


## CDP - 1HPI - Bystander vs Mock-Infected
BysvsMock.mdp.1hpi.dotplot <- DotPlot(
  object = subset(sc.seurat, 
                  progenitor=='MDP' & 
                    plate_number=='P193' & 
                    Bystander != 'Infected'), 
  features = c(head(degs.filtered$bysvsMock.mdp.deg$gene, n_markers),
               tail(degs.filtered$bysvsMock.mdp.deg$gene, n_markers))
) + coord_flip() +
  scale_color_viridis() +
  labs(x='', y='') +
  theme(axis.text.x = element_text(angle = 45, hjust = 1)) +
  ggtitle('MDP - 1HPI - Infected vs Mock')


###################################################################################
## 1 HPI - Dot plots
gridExtra::grid.arrange(
  infvsBys.cdp.1hpi.dotplot,
  infvsMock.cdp.1hpi.dotplot,
  BysvsMock.cdp.1hpi.dotplot, 
  infvsBys.mdp.1hpi.dotplot,
  infvsMock.mdp.1hpi.dotplot,
  BysvsMock.mdp.1hpi.dotplot,
  ncol=3
)
```


### 24 hpi (Plate 196)


```{r, fig.height=16, fig.width=16}
# CDP - 24HPI - Infected vs Bystander
infvsBys.cdp.24hpi.dotplot <- DoHeatmap(
  object = subset(sc.seurat, 
                  progenitor=='CDP' & 
                    plate_number=='P196' & 
                    Bystander != 'Mock-Infected'), 
  features = c(head(degs.filtered$infvsBys.cdp.deg.24$gene, n_markers),
               tail(degs.filtered$infvsBys.cdp.deg.24$gene, n_markers)), 
  draw.lines = FALSE, 
  size = 3, 
  angle = 0, combine = TRUE
) + scale_fill_viridis() +
  theme(legend.position = 'none') +
  ggtitle('CDP - 24HPI - Infected vs Bystander')


# CDP - 24HPI - Infected vs Mock
infvsMock.cdp.24hpi.dotplot <- DoHeatmap(
  object = subset(sc.seurat, 
                  progenitor=='CDP' & 
                    plate_number=='P196' & 
                    Bystander != 'Bystander'), 
  features = c(head(degs.filtered$infvsMock.cdp.deg.24$gene, n_markers),
               tail(degs.filtered$infvsMock.cdp.deg.24$gene, n_markers)), 
  draw.lines = FALSE, 
  size = 3, 
  angle = 0, combine = TRUE
) + scale_fill_viridis() +
  theme(legend.position = 'none') +
  ggtitle('CDP - 24HPI - Infected vs Mock')

##############################################################################################

## CDP - 24HPI - Infected vs Bystander
infvsBys.cdp.24hpi.dotplot <- DotPlot(
  object = subset(sc.seurat, 
                  progenitor=='CDP' & 
                    plate_number=='P196' & 
                    Bystander != 'Mock-Infected'), 
  features = c(head(degs.filtered$infvsBys.cdp.deg.24$gene, n_markers),
               tail(degs.filtered$infvsBys.cdp.deg.24$gene, n_markers))
) + coord_flip() +
  scale_color_viridis() +
  labs(x='', y='') +
  theme(axis.text.x = element_text(angle = 45, hjust = 1)) +
  ggtitle('CDP - 24HPI - Infected vs Bystander')


## CDP - 24HPI - Infected vs Mock-Infected
infvsMock.cdp.24hpi.dotplot <- DotPlot(
  object = subset(sc.seurat, 
                  progenitor=='CDP' & 
                    plate_number=='P196' & 
                    Bystander != 'Bystander'), 
  features = c(head(degs.filtered$infvsMock.cdp.deg.24$gene, n_markers),
               tail(degs.filtered$infvsMock.cdp.deg.24$gene, n_markers))
) + coord_flip() +
  scale_color_viridis() +
  labs(x='', y='') +
  theme(axis.text.x = element_text(angle = 45, hjust = 1)) +
  ggtitle('CDP - 24HPI - Infected vs Mock')


## CDP - 24HPI - Bystander vs Mock-Infected
BysvsMock.cdp.24hpi.dotplot <- DotPlot(
  object = subset(sc.seurat, 
                  progenitor=='CDP' & 
                    plate_number=='P196' & 
                    Bystander != 'Infected'), 
  features = degs.filtered$bysvsMock.cdp.deg.24$gene
) + coord_flip() +
  scale_color_viridis() +
  labs(x='', y='') +
  theme(axis.text.x = element_text(angle = 45, hjust = 1)) +
  ggtitle('CDP - 24HPI - Infected vs Mock')

###################################################################################

## CDP - 24HPI - Infected vs Bystander
infvsBys.mdp.24hpi.dotplot <- DotPlot(
  object = subset(sc.seurat, 
                  progenitor=='MDP' & 
                    plate_number=='P196' & 
                    Bystander != 'Mock-Infected'), 
  features = degs.filtered$infvsBys.mdp.deg.24$gene
) + coord_flip() +
  scale_color_viridis() +
  labs(x='', y='') +
  theme(axis.text.x = element_text(angle = 45, hjust = 1)) +
  ggtitle('MDP - 24HPI - Infected vs Bystander')


## CDP - 24HPI - Infected vs Mock-Infected
infvsMock.mdp.24hpi.dotplot <- DotPlot(
  object = subset(sc.seurat, 
                  progenitor=='MDP' & 
                    plate_number=='P196' & 
                    Bystander != 'Bystander'), 
  features = c(head(degs.filtered$infvsMock.mdp.deg.24$gene, n_markers),
               tail(degs.filtered$infvsMock.mdp.deg.24$gene, n_markers))
) + coord_flip() +
  scale_color_viridis() +
  labs(x='', y='') +
  theme(axis.text.x = element_text(angle = 45, hjust = 1)) +
  ggtitle('MDP - 24HPI - Infected vs Mock')


## CDP - 24HPI - Bystander vs Mock-Infected
BysvsMock.mdp.24hpi.dotplot <- DotPlot(
  object = subset(sc.seurat, 
                  progenitor=='MDP' & 
                    plate_number=='P196' & 
                    Bystander != 'Infected'), 
  features = c(head(degs.filtered$bysvsMock.mdp.deg.24$gene, n_markers),
               tail(degs.filtered$bysvsMock.mdp.deg.24$gene, n_markers))
) + coord_flip() +
  scale_color_viridis() +
  labs(x='', y='') +
  theme(axis.text.x = element_text(angle = 45, hjust = 1)) +
  ggtitle('MDP - 24HPI - Bystander vs Mock')


###################################################################################
## 1 HPI - Dot plots
gridExtra::grid.arrange(
  infvsBys.cdp.24hpi.dotplot,
  infvsMock.cdp.24hpi.dotplot,
  BysvsMock.cdp.24hpi.dotplot, 
  infvsBys.mdp.24hpi.dotplot,
  infvsMock.mdp.24hpi.dotplot,
  BysvsMock.mdp.24hpi.dotplot,
  ncol=3
)
```


# GSEA {.tabset}

We performed a gene set enrichment analysis using all DEGs for each condition. The
enriched gene sets are shown in the next plots. The dataset MSigDB_Hallmark_2020
was used for the analysis.


```{r, GSEA, fig.align='center', results='hide'}
gsea.list <- lapply(degs.filtered, 
       function(x) {
         degs <- dplyr::select(x, gene) %>%
                      unlist()
         enrichr(degs, databases = "MSigDB_Hallmark_2020")
       }
)


gsea.plot <- lapply(names(gsea.list), 
                       function(x) 
                         plotEnrich(gsea.list[x][[1]]$MSigDB_Hallmark_2020, 
                                    showTerms = 20, 
                                    numChar = 100, 
                                    y = "Count", 
                                    orderBy = "Count") +
                              scale_fill_viridis() +
                              xlab('') +
                              ggtitle(x)
)
names(gsea.plot) <- names(gsea.list)
```


```{r, GSEA_plot, results = "asis", fig.align='center', eval=FALSE}
names(gsea.plot) <- c('Infected vs Bystander - CDP - 1HPI',
                      'Infected vs Mock - CDP - 1HPI',
                      'Bystander vs Mock - CDP - 1HPI',
                      'Infected vs Bystander - MDP - 1HPI',
                      'Infected vs Mock - MDP - 1HPI',
                      'Bystander vs Mock - MDP - 1HPI',
                      'Infected vs Bystander - CDP - 24HPI',
                      'Infected vs Mock - CDP - 24HPI',
                      'Bystander vs Mock - CDP - 24HPI',
                      'Infected vs Bystander - MDP - 24HPI',
                      'Infected vs Mock - MDP - 24HPI',
                      'Bystander vs Mock - MDP - 24HPI')
for (i in 1:length(gsea.plot)) {
    cat(paste0('## ', names(gsea.plot)[i], '\n\n'))
    print(gsea.plot[i][[1]])
    cat(' \n\n')
}
```

## GSEA (1 hpi)


The next barplot show the enrichment analysis of the DEGs obtained above at time
1 hpi.


```{r, fig.height=10, fig.width=17}
gridExtra::grid.arrange(grobs=gsea.plot[1:6], ncol=3)
```

## GSEA (24 hpi)


The next barplot show the enrichment analysis of the DEGs obtained above at
24 hpi. 

```{r, fig.height=10, fig.width=17}
gridExtra::grid.arrange(grobs=gsea.plot[7:12], ncol=3)
```

We can conclude from the GSEA that immune responses change between infectious 
status.


# Ordering cells

We were interested in the dynamical behavior of cells during the infection 
process. Then, we projected the cells into diffusion maps and ordered them
according to pseudotime. The next diffusion maps show the projection of 
CDP and MDP derived DCs at 1 hpi (top left and right) and CDP and MDP
derived DCs at 24 hpi (bottom left and right).
To calculate the difussion maps significant DEGs between Infected, Bystander 
and  Mock-Infected cells for the corresponding subset of cells were used. 
The destiny R package were used with parameters n_pcs=50 and n_eigs=50.
The dimensions DC1 and DC2 represent the first two eigenvectors of the 
Markovian transition matrix.

NOTE: here the set of DEGs are calculated in a different manner from 
the DEA above. For the categories Infected, Bystander and Mock-Infected
the categories are tested each *vs* the rest and then DEGs are merged.

```{r, fig.width=10, fig.height=8}
## 1 hpi - CDP
cdp.1hpi <- subset(sc.seurat,
                      plate_number == 'P193' &
                        progenitor == 'CDP')
Idents(cdp.1hpi) <- cdp.1hpi$Bystander
degs.1hpi <- FindAllMarkers(cdp.1hpi)
degs.1hpi <- subset(degs.1hpi, p_val < 0.05)
eset <- ExpressionSet(
  assayData = cdp.1hpi@assays$RNA@scale.data,
  phenoData = AnnotatedDataFrame(cdp.1hpi@meta.data)
)

difmap.1hpi <- DiffusionMap(eset, 
                       vars = degs.1hpi$gene,
                       n_pcs = 50, 
                       n_eigs = 50)

res.cdp.1hpi <- cbind(difmap.1hpi@data_env$data@phenoData@data, 
             DC1=difmap.1hpi$DC1,
             DC2=difmap.1hpi$DC2,
             DC3=difmap.1hpi$DC3,
             DC4=difmap.1hpi$DC4) 


cdp.1hpi <- res.cdp.1hpi %>% ggplot(aes(x=DC1, y=-DC2,
                   colour=Bystander)) +
        geom_point(size=2.5, alpha=0.7) + 
        theme_classic() +
        labs(colour='') +
        ggtitle('CDP - 1 hpi')

#####################################################################
## 1 hpi - MDP
mdp.1hpi <- subset(sc.seurat,
                      plate_number == 'P193' &
                        progenitor == 'MDP')
Idents(mdp.1hpi) <- mdp.1hpi$Bystander
degs.1hpi <- FindAllMarkers(mdp.1hpi)
degs.1hpi <- subset(degs.1hpi, p_val < 0.05)
eset <- ExpressionSet(
  assayData = mdp.1hpi@assays$RNA@scale.data,
  phenoData = AnnotatedDataFrame(mdp.1hpi@meta.data)
)

difmap.1hpi <- DiffusionMap(eset, 
                       vars = degs.1hpi$gene,
                       n_pcs = 50, 
                       n_eigs = 50)

res.mdp.1hpi <- cbind(difmap.1hpi@data_env$data@phenoData@data, 
             DC1=difmap.1hpi$DC1,
             DC2=difmap.1hpi$DC2,
             DC3=difmap.1hpi$DC3,
             DC4=difmap.1hpi$DC4) 

mdp.1hpi <- res.mdp.1hpi %>% ggplot(aes(x=DC1, y=DC2,
                   colour=Bystander)) +
        geom_point(size=2.5, alpha=0.7) +
        theme_classic() +
        labs(colour='') +
        ggtitle('MDP - 1 hpi')


#########################################################################

## 24 hpi - CDP
cdp.24hpi <- subset(sc.seurat,
                      plate_number == 'P196' &
                        progenitor == 'CDP')
Idents(cdp.24hpi) <- cdp.24hpi$Bystander
degs.24hpi <- FindAllMarkers(cdp.24hpi)
degs.24hpi <- subset(degs.24hpi, p_val < 0.05)
eset <- ExpressionSet(
  assayData = cdp.24hpi@assays$RNA@scale.data,
  phenoData = AnnotatedDataFrame(cdp.24hpi@meta.data)
)

difmap.24hpi <- DiffusionMap(eset, 
                       vars = degs.24hpi$gene,
                       n_pcs = 50, 
                       n_eigs = 50)

res.cdp.24hpi <- cbind(difmap.24hpi@data_env$data@phenoData@data, 
             DC1=difmap.24hpi$DC1,
             DC2=difmap.24hpi$DC2,
             DC3=difmap.24hpi$DC3,
             DC4=difmap.24hpi$DC4) 


cdp.24hpi <- res.cdp.24hpi %>% ggplot(aes(x=DC1, y=DC2,
                   colour=Bystander)) +
        geom_point(size=2.5, alpha=0.7) + 
        theme_classic() +
        labs(colour='') +
        ggtitle('CDP - 24 hpi')

#####################################################################
## 24 hpi - MDP
mdp.24hpi <- subset(sc.seurat,
                      plate_number == 'P196' &
                        progenitor == 'MDP')
Idents(mdp.24hpi) <- mdp.24hpi$Bystander
degs.24hpi <- FindAllMarkers(mdp.24hpi)
degs.24hpi <- subset(degs.24hpi, p_val < 0.05)
eset <- ExpressionSet(
  assayData = mdp.24hpi@assays$RNA@scale.data,
  phenoData = AnnotatedDataFrame(mdp.24hpi@meta.data)
)

difmap.24hpi <- DiffusionMap(eset, 
                       vars = degs.24hpi$gene,
                       n_pcs = 23, 
                       n_eigs = 23, 
                       k = 30)

res.mdp.24hpi <- cbind(difmap.24hpi@data_env$data@phenoData@data, 
             DC1=difmap.24hpi$DC1,
             DC2=difmap.24hpi$DC2,
             DC3=difmap.24hpi$DC3,
             DC4=difmap.24hpi$DC4) 

my_theme <- theme_bw() +
                theme(panel.grid = element_blank()) 

mdp.24hpi <- res.mdp.24hpi %>% ggplot(aes(x=DC1, y=-DC2,
                   colour=Bystander)) +
        geom_point(size=2.5, alpha=0.7) +
        theme_classic() +
        labs(colour='') +
        ggtitle('MDP - 24 hpi')


gridExtra::grid.arrange(
  cdp.1hpi,
  mdp.1hpi,
  cdp.24hpi,
  mdp.24hpi,
  ncol=2
)
```


From this analysis, two clear branches can be observed for Infected and Bystander cells
which suggest different underlying signatures in each subsets. Bystander cells are more
similar to Mock-Infected control cells as expected.


## Checking infection status


The next diffusion maps show the infection status by the bacteria attached category as
a double check of the previous analysis.


```{r, fig.width=10, fig.height=8}
cdp.1hpi <- res.cdp.1hpi %>% ggplot(aes(x=DC1, y=-DC2,
                   colour=bacteria_attached)) +
        geom_point(size=2.5, alpha=0.7) + 
        theme_classic() +
        labs(colour='') +
        ggtitle('CDP - 1 hpi')


mdp.1hpi <- res.mdp.1hpi %>% ggplot(aes(x=DC1, y=DC2,
                   colour=bacteria_attached)) +
        geom_point(size=2.5, alpha=0.7) +
        theme_classic() +
        labs(colour='') +
        ggtitle('MDP - 1 hpi')


cdp.24hpi <- res.cdp.24hpi %>% ggplot(aes(x=DC1, y=DC2,
                   colour=bacteria_attached)) +
        geom_point(size=2.5, alpha=0.7) + 
        theme_classic() +
        labs(colour='') +
        ggtitle('CDP - 24 hpi')



mdp.24hpi <- res.mdp.24hpi %>% ggplot(aes(x=DC1, y=-DC2,
                   colour=bacteria_attached)) +
        geom_point(size=2.5, alpha=0.7) +
        theme_classic() +
        labs(colour='') +
        ggtitle('MDP - 24 hpi')


gridExtra::grid.arrange(
  cdp.1hpi,
  mdp.1hpi,
  cdp.24hpi,
  mdp.24hpi,
  ncol=2
)
```



## Transcription factor activity {.tabset}


Then, Transcription factor activities (TFA) were correlated to the pseudotime in the 
diffusion maps. We focused in the Diffusion Component 1 (DC1) since it can
be seen that Infected, Bystander and Mock-Infected are ordered along this
axis. The following joystick plots show the spearman correlations between TF 
activities and DC1. 


```{r, eval=FALSE}
## Normalizing and saving data
dir.create('data/scenic_input')
sc.seurat <- SCTransform(sc.seurat, variable.features.n = 4000)
sc.seurat@assays$SCT@scale.data %>%
  t() %>%
  write.table('data/scenic_input/DC_single_cells.tsv', 
              sep = '\t')
```



```{r, loading_scenic_results}
tfa <- read.csv('analysis/scenic/aucell.csv') 
## Collecting Diffusion maps 
dif.maps <- list(res.cdp.1hpi=res.cdp.1hpi,
                 res.mdp.1hpi=res.mdp.1hpi,
                 res.cdp.24hpi=res.cdp.24hpi,
                 res.mdp.24hpi=res.mdp.24hpi)

comps <- c('DC1', 'DC2', 'DC3', 'DC4')
tfs <- grep('\\.', colnames(tfa), value = TRUE)
cor.dc1.plots <- lapply(names(dif.maps), 
       function(x) {
         df <- add_rownames(dif.maps[x][[1]], var = 'Cell')
         tfa.pstime <- merge(df, tfa, by='Cell')
         mtx <- tfa.pstime[, c(comps, tfs)]
         mtx.cor <- cor(mtx, method = 'spearman')
         mtx.cor %>%
          as.data.frame() %>%
           arrange(desc(abs(DC1))) %>%
           add_rownames(var='TF') %>%
           dplyr::filter(! TF %in% c(comps)) %>%
           dplyr::mutate(rank=1:n()) %>%
           dplyr::mutate(TF=gsub('\\.', '', TF)) %>%
           dplyr::mutate(highlight=ifelse(rank<80, TRUE, FALSE)) %>%
           dplyr::mutate(gene_label=ifelse(highlight==TRUE, TF, '')) %>%
           ggplot(aes(x=rank, y=abs(DC1),
                      colour=highlight, 
                      label=gene_label)) +
                  geom_point() +
                  geom_text_repel() +
                  theme_bw() +
                  theme(legend.position = 'none', 
                        panel.grid = element_blank()) +
                  labs(x='Rank', y='Abs(Cor(TFA, DC1))')+
                  ggtitle(x)
       }
)
names(cor.dc1.plots) <- gsub('red', '', names(dif.maps))
```


```{r, results='asis'}
for (name in names(cor.dc1.plots)) {
  cat('### ', name, '\n\n')
  print(cor.dc1.plots[name][[1]])
  cat(' \n\n')
}
```


We can observe, that for example Spi1 and Jun are TFs present in the top
most correlated to DC1. 

## TFA projection on pseudotime


We wanted to visualize the dynamic changes of TFA across the cells. Then, TFA were
projected over the previously inferred pseudotime trajectories. The next Diffusion 
maps show the expression of Jun along the trajectories.

## Spi1 TFA

```{r, fig.width=10, fig.height=8}
pseudo.tfa.list <- lapply(names(dif.maps), 
       function(x) {
         df <- add_rownames(dif.maps[x][[1]], var = 'Cell')
         tfa.pstime <- merge(df, tfa, by='Cell')
         mtx <- dplyr::select(tfa.pstime, DC1:`Zfp597...`)
         mtx 
       }
)
names(pseudo.tfa.list) <- gsub('res.', '', names(dif.maps))

cdp.1hp.sp1 <- pseudo.tfa.list[1][[1]] %>%
  ggplot(aes(x=DC1, y=-DC2,
             colour=`Spi1...`)) +
      geom_point(size=3) +
      scale_color_viridis() +
      theme_classic() +
      ggtitle(names(pseudo.tfa.list)[1])

mdp.1hp.sp1 <- pseudo.tfa.list[2][[1]] %>%
  ggplot(aes(x=DC1, y=DC2,
             colour=`Spi1...`)) +
      geom_point(size=3) +
      scale_color_viridis() +
      theme_classic() +
      ggtitle(names(pseudo.tfa.list)[2])

cdp.24hp.sp1 <- pseudo.tfa.list[3][[1]] %>%
  ggplot(aes(x=DC1, y=DC2,
             colour=`Spi1...`)) +
      geom_point(size=3) +
      scale_color_viridis() +
      theme_classic() +
      ggtitle(names(pseudo.tfa.list)[3])

mdp.24hp.sp1 <- pseudo.tfa.list[4][[1]] %>%
  ggplot(aes(x=DC1, y=-DC2,
             colour=`Spi1...`)) +
      geom_point(size=3) +
      scale_color_viridis() +
      theme_classic() +
      ggtitle(names(pseudo.tfa.list)[4])

gridExtra::grid.arrange(
  cdp.1hp.sp1,
  mdp.1hp.sp1,
  cdp.24hp.sp1,
  mdp.24hp.sp1,
  ncol=2
)
```


## Jun TFA


```{r, fig.width=10, fig.height=8}
pseudo.tfa.list <- lapply(names(dif.maps), 
       function(x) {
         df <- add_rownames(dif.maps[x][[1]], var = 'Cell')
         tfa.pstime <- merge(df, tfa, by='Cell')
         mtx <- dplyr::select(tfa.pstime, DC1:`Zfp597...`)
         mtx 
       }
)
names(pseudo.tfa.list) <- gsub('res.', '', names(dif.maps))

cdp.1hp.jun <- pseudo.tfa.list[1][[1]] %>%
  ggplot(aes(x=DC1, y=-DC2,
             colour=`Jun...`)) +
      geom_point(size=3) +
      scale_color_viridis() +
      theme_classic() +
      ggtitle(names(pseudo.tfa.list)[1])

mdp.1hp.jun <- pseudo.tfa.list[2][[1]] %>%
  ggplot(aes(x=DC1, y=DC2,
             colour=`Jun...`)) +
      geom_point(size=3) +
      scale_color_viridis() +
      theme_classic() +
      ggtitle(names(pseudo.tfa.list)[2])

cdp.24hp.jun <- pseudo.tfa.list[3][[1]] %>%
  ggplot(aes(x=DC1, y=DC2,
             colour=`Jun...`)) +
      geom_point(size=3) +
      scale_color_viridis() +
      theme_classic() +
      ggtitle(names(pseudo.tfa.list)[3])

mdp.24hp.jun <- pseudo.tfa.list[4][[1]] %>%
  ggplot(aes(x=DC1, y=-DC2,
             colour=`Jun...`)) +
      geom_point(size=3) +
      scale_color_viridis() +
      theme_classic() +
      ggtitle(names(pseudo.tfa.list)[4])

gridExtra::grid.arrange(
  cdp.1hp.jun,
  mdp.1hp.jun,
  cdp.24hp.jun,
  mdp.24hp.jun,
  ncol=2
)
```


It can be seen that Jun is more active in the infected cells compared to bystander
and non-infected mock cells.


# Comparing CDP vs MDP responses 

We compared responses in CDP vs MDP derived DCs. DEA was performed for infected,
bystander and mock-infected cells at 1 and 24 hpi separately.

```{r cdp_vs_mdp, fig.width=12, fig.height=9}
Idents(sc.seurat) <- as.character(sc.seurat$progenitor)

## Mock 1hpi
prog.mock.193 <- sc.seurat %>%
  subset(plate_number=='P193') %>%
  subset(Bystander != 'Mock-Infected') %>%
  FindMarkers(ident.1='CDP') %>%
  arrange(desc(abs(avg_log2FC))) %>%
  add_rownames('gene') %>%
  mutate(selected_gene=(toupper(gene) %in% toupper(selected_genes))) %>%
  filter(!grepl('^Gm', gene)) %>%
  mutate(highlight=ifelse( ( abs(avg_log2FC) > 0.9  & p_val < 0.05),
                           TRUE, FALSE) ) %>%
  mutate(label=ifelse(highlight==TRUE | 
                        selected_gene ==TRUE, gene, ''))
prog.mock.193.plot <- prog.mock.193 %>%  
  ggplot(aes(x=avg_log2FC, y=-log10(p_val),
             colour=highlight,
             label=label)) +
        geom_point(show.legend = FALSE) +
        geom_point(data=subset(prog.mock.193,
                               selected_gene==TRUE), 
                   aes(colour='green'),
                   show.legend=FALSE) +
        geom_text_repel(show.legend=FALSE, 
                        colour='black', 
                        force = 3.5, 
                        max.overlaps = 200) +
        theme(legend.position = 'none') +
        scale_color_manual(values = c('salmon', 'green', 
                                      'black', 'red')) +
        theme_classic() +
        xlim(-2, 2) +
        ggtitle('CDP vs MDP - 1HPI - Mock-Infected')


## Bystander 1hpi
prog.byst.193 <- sc.seurat %>%
  subset(plate_number=='P193') %>%
  subset(Bystander != 'Bystander') %>%
  FindMarkers(ident.1='CDP') %>%
  arrange(desc(abs(avg_log2FC))) %>%
  add_rownames('gene') %>%
  mutate(selected_gene=(toupper(gene) %in% toupper(selected_genes))) %>%
  filter(!grepl('^Gm', gene)) %>%
  mutate(highlight=ifelse( ( abs(avg_log2FC) > 0.9  & p_val < 0.05),
                           TRUE, FALSE) ) %>%
  mutate(label=ifelse(highlight==TRUE | 
                        selected_gene ==TRUE, gene, ''))
prog.byst.193.plot <- prog.byst.193 %>%  
  ggplot(aes(x=avg_log2FC, y=-log10(p_val),
             colour=highlight,
             label=label)) +
        geom_point(show.legend = FALSE) +
        geom_point(data=subset(prog.byst.193,
                               selected_gene==TRUE), 
                   aes(colour='green'),
                   show.legend=FALSE) +
        geom_text_repel(show.legend=FALSE, 
                        colour='black', 
                        force = 3.5, 
                        max.overlaps = 200) +
        theme(legend.position = 'none') +
        scale_color_manual(values = c('salmon', 'green', 
                                      'black', 'red')) +
        theme_classic() +
        xlim(-2, 2) +
        ggtitle('CDP vs MDP - 1HPI - Bystander')


## Infected 1hpi
prog.inf.193 <- sc.seurat %>%
  subset(plate_number=='P193') %>%
  subset(Bystander != 'Infected') %>%
  FindMarkers(ident.1='CDP') %>%
  arrange(desc(abs(avg_log2FC))) %>%
  add_rownames('gene') %>%
  mutate(selected_gene=(toupper(gene) %in% toupper(selected_genes))) %>%
  filter(!grepl('^Gm', gene)) %>%
  mutate(highlight=ifelse( ( abs(avg_log2FC) > 0.9  & p_val < 0.05),
                           TRUE, FALSE) ) %>%
  mutate(label=ifelse(highlight==TRUE | 
                        selected_gene ==TRUE, gene, ''))
prog.inf.193.plot <- prog.inf.193 %>%  
  ggplot(aes(x=avg_log2FC, y=-log10(p_val),
             colour=highlight,
             label=label)) +
        geom_point(show.legend = FALSE) +
        geom_point(data=subset(prog.inf.193,
                               selected_gene==TRUE), 
                   aes(colour='green'),
                   show.legend=FALSE) +
        geom_text_repel(show.legend=FALSE, 
                        colour='black', 
                        force = 3.5, 
                        max.overlaps = 200) +
        theme(legend.position = 'none') +
        scale_color_manual(values = c('salmon', 'green', 
                                      'black', 'red')) +
        theme_classic() +
        xlim(-2, 2) +
        ggtitle('CDP vs MDP - 1HPI - Infected')


################################################################################

## Mock 24hpi
prog.mock.196 <- sc.seurat %>%
  subset(plate_number=='P196') %>%
  subset(Bystander != 'Mock-Infected') %>%
  FindMarkers(ident.1='CDP') %>%
  arrange(desc(abs(avg_log2FC))) %>%
  add_rownames('gene') %>%
  mutate(selected_gene=(toupper(gene) %in% toupper(selected_genes))) %>%
  filter(!grepl('^Gm', gene)) %>%
  mutate(highlight=ifelse( ( abs(avg_log2FC) > 1.5  & p_val < 0.0005),
                           TRUE, FALSE) ) %>%
  mutate(label=ifelse(highlight==TRUE | 
                        selected_gene ==TRUE, gene, ''))
prog.mock.196.plot <- prog.mock.196 %>%  
  ggplot(aes(x=avg_log2FC, y=-log10(p_val),
             colour=highlight,
             label=label)) +
        geom_point(show.legend = FALSE) +
        geom_point(data=subset(prog.mock.196,
                               selected_gene==TRUE), 
                   aes(colour='green'),
                   show.legend=FALSE) +
        geom_text_repel(show.legend=FALSE, 
                        colour='black', 
                        force = 3.5, 
                        max.overlaps = 200) +
        theme(legend.position = 'none') +
        scale_color_manual(values = c('salmon', 'green', 
                                      'black', 'red')) +
        theme_classic() +
        xlim(-2, 2) +
        ggtitle('CDP vs MDP - 24HPI - Mock-Infected')


## Bystander 1hpi
prog.byst.196 <- sc.seurat %>%
  subset(plate_number=='P196') %>%
  subset(Bystander != 'Bystander') %>%
  FindMarkers(ident.1='CDP') %>%
  arrange(desc(abs(avg_log2FC))) %>%
  add_rownames('gene') %>%
  mutate(selected_gene=(toupper(gene) %in% toupper(selected_genes))) %>%
  filter(!grepl('^Gm', gene)) %>%
  mutate(highlight=ifelse( ( abs(avg_log2FC) > 1.5  & p_val < 0.005),
                           TRUE, FALSE) ) %>%
  mutate(label=ifelse(highlight==TRUE | 
                        selected_gene ==TRUE, gene, ''))
prog.byst.196.plot <- prog.byst.196 %>%  
  ggplot(aes(x=avg_log2FC, y=-log10(p_val),
             colour=highlight,
             label=label)) +
        geom_point(show.legend = FALSE) +
        geom_point(data=subset(prog.byst.196,
                               selected_gene==TRUE), 
                   aes(colour='green'),
                   show.legend=FALSE) +
        geom_text_repel(show.legend=FALSE, 
                        colour='black', 
                        force = 3.5, 
                        max.overlaps = 200) +
        theme(legend.position = 'none') +
        scale_color_manual(values = c('salmon', 'green', 
                                      'black', 'red')) +
        theme_classic() +
        xlim(-2, 2) +
        ggtitle('CDP vs MDP - 24HPI - Bystander')


## Infected 24hpi
prog.inf.196 <- sc.seurat %>%
  subset(plate_number=='P196') %>%
  subset(Bystander != 'Infected') %>%
  FindMarkers(ident.1='CDP') %>%
  arrange(desc(abs(avg_log2FC))) %>%
  add_rownames('gene') %>%
  mutate(selected_gene=(toupper(gene) %in% toupper(selected_genes))) %>%
  filter(!grepl('^Gm', gene)) %>%
  mutate(highlight=ifelse( ( abs(avg_log2FC) > 1.75  & p_val < 0.005),
                           TRUE, FALSE) ) %>%
  mutate(label=ifelse(highlight==TRUE | 
                        selected_gene ==TRUE, gene, ''))
prog.inf.196.plot <- prog.inf.196 %>%  
  ggplot(aes(x=avg_log2FC, y=-log10(p_val),
             colour=highlight,
             label=label)) +
        geom_point(show.legend = FALSE) +
        geom_point(data=subset(prog.inf.196,
                               selected_gene==TRUE), 
                   aes(colour='green'),
                   show.legend=FALSE) +
        geom_text_repel(show.legend=FALSE, 
                        colour='black', 
                        force = 3.5, 
                        max.overlaps = 200) +
        theme(legend.position = 'none') +
        scale_color_manual(values = c('salmon', 'green', 
                                      'black', 'red')) +
        theme_classic() +
        xlim(-2, 2) +
        ggtitle('CDP vs MDP - 24HPI - Infected')

gridExtra::grid.arrange(
  prog.inf.193.plot,
  prog.byst.193.plot,
  prog.mock.193.plot,
  prog.inf.196.plot,
  prog.byst.196.plot,
  prog.mock.196.plot,
  ncol=3
)

```


# Additional tables

The next tables exhaustively list all the genes in the DEGs for each comparison having 
p-val < 0.05.

## 1 HPI (Plate 193)

### CDPs. 

* Table comparing Infected *vs* Bystander

```{r table_cdp_inf_vs_bys_1hpi}
infvsBys.cdp.deg %>%
  filter(p_val < 0.05) %>%
  paged_table()
```

* Table comparing Bystander *vs* Mock-Infected

```{r}
bysvsMock.cdp.deg %>%
  filter(p_val < 0.05) %>%
  paged_table()
```


* Table comparing Infected *vs* Mock-Infected


```{r}
infvsMock.cdp.deg %>%
  filter(p_val < 0.05) %>%
  paged_table()
```


### MDPs. 

* Table comparing Infected *vs* Bystander

```{r table_mdp_inf_vs_bys_1hpi}
infvsBys.mdp.deg %>%
  filter(p_val < 0.05) %>%
  paged_table()
```

* Table comparing Bystander *vs* Mock-Infected

```{r}
bysvsMock.mdp.deg %>%
  filter(p_val < 0.05) %>%
  paged_table()
```


* Table comparing Infected *vs* Mock-Infected

```{r}
infvsMock.mdp.deg %>%
  filter(p_val < 0.05) %>%
  paged_table()
```


## 24 HPI (Plate 196)


### CDPs. 

* Table comparing Infected *vs* Bystander

```{r table_cdp_inf_vs_bys_24hpi}
infvsBys.cdp.deg.24 %>%
  filter(p_val < 0.05) %>%
  paged_table()
```

* Table comparing Bystander *vs* Mock-Infected

```{r}
bysvsMock.cdp.deg.24 %>%
  filter(p_val < 0.05) %>%
  paged_table()
```


* Table comparing Infected *vs* Mock-Infected


```{r}
infvsMock.cdp.deg.24 %>%
  filter(p_val < 0.05) %>%
  paged_table()
```


### MDPs. 

* Table comparing Infected *vs* Bystander

```{r table_mdp_inf_vs_bys_24hpi}
infvsBys.mdp.deg.24 %>%
  filter(p_val < 0.05) %>%
  paged_table()
```

* Table comparing Bystander *vs* Mock-Infected

```{r}
bysvsMock.mdp.deg.24 %>%
  filter(p_val < 0.05) %>%
  paged_table()
```


* Table comparing Infected *vs* Mock-Infected

```{r}
infvsMock.mdp.deg.24 %>%
  filter(p_val < 0.05) %>%
  paged_table()
```
